File: app/src/main/java/com/semye/android/WifiMainActivity.java
package com.semye.android;

import android.content.Context;
import android.net.DhcpInfo;
import android.net.wifi.ScanResult;
import android.net.wifi.WifiConfiguration;
import android.net.wifi.WifiInfo;
import android.net.wifi.WifiManager;
import android.net.wifi.hotspot2.PasspointConfiguration;
import android.os.Build;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import java.util.List;

public class WifiMainActivity extends AppCompatActivity {

    private WifiManager wifiManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_wifi);
        wifiManager = (WifiManager) getApplicationContext().getSystemService(Context.WIFI_SERVICE);
        show();
    }


    private void show() {
        //获取当前的wifi网络配置
        for (WifiConfiguration configuration : wifiManager.getConfiguredNetworks()) {
            System.out.println(configuration.toString());
        }

        WifiInfo connectionInfo = wifiManager.getConnectionInfo();
        System.out.println(connectionInfo.toString());

        DhcpInfo dhcpInfo = wifiManager.getDhcpInfo();
        System.out.println(dhcpInfo.toString());


        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
            System.out.println("isScanAlwaysAvailable" + wifiManager.isScanAlwaysAvailable());
        }

        System.out.println("isWifiEnabled" + wifiManager.isWifiEnabled());
        System.out.println("wifiState" + wifiManager.getWifiState());


        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            System.out.println("5G" + wifiManager.is5GHzBandSupported());
            System.out.println("isDeviceToApRttSupported" + wifiManager.isDeviceToApRttSupported());
            System.out.println("isEnhancedPowerReportingSupported" + wifiManager.isEnhancedPowerReportingSupported());
            System.out.println("isP2pSupported" + wifiManager.isP2pSupported());
            System.out.println("isPreferredNetworkOffloadSupported" + wifiManager.isPreferredNetworkOffloadSupported());
            System.out.println("isTdlsSupported" + wifiManager.isTdlsSupported());
        }

        List<ScanResult> scanResults = wifiManager.getScanResults();
        for (ScanResult result : scanResults) {
            System.out.println(result.toString());
        }

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            try {
                List<PasspointConfiguration> passpointConfigurations = wifiManager.getPasspointConfigurations();
                for (PasspointConfiguration passpointConfiguration : passpointConfigurations) {
                    System.out.println(passpointConfiguration.toString());
                }
            } catch (Exception e) {
                e.printStackTrace();
            }
        }

    }
}

File: app/src/main/java/com/semye/android/network/wifi/WifiMainActivity.kt
package com.semye.android.network.wifi

import android.Manifest
import android.content.pm.PackageManager
import android.net.wifi.WifiManager
import android.os.Build
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.semye.android.R

class WifiMainActivity : AppCompatActivity() {

    private var wifiManager: WifiManager? = null

    

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main_wifi)
        wifiManager = applicationContext.getSystemService(WIFI_SERVICE) as? WifiManager

        //Q版本后不允许打开关闭wifi
        findViewById<Button>(R.id.open).setOnClickListener {
            wifiManager?.isWifiEnabled = true
        }
        findViewById<Button>(R.id.close).setOnClickListener {
            wifiManager?.isWifiEnabled = false
        }

        findViewById<TextView>(R.id.info).text = wifiManager?.connectionInfo.toString()
        findViewById<TextView>(R.id.dhcpinfo).text = wifiManager?.dhcpInfo.toString()






        show()
    }

    private fun show() {
        //获取当前的wifi网络配置
        if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED
            || ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_WIFI_STATE) == PackageManager.PERMISSION_GRANTED) {
            for (configuration in wifiManager!!.configuredNetworks) {
                println(configuration.toString())
            }
        }

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
            println("isScanAlwaysAvailable" + wifiManager!!.isScanAlwaysAvailable)
        }
        println("isWifiEnabled" + wifiManager!!.isWifiEnabled)
        println("wifiState" + wifiManager!!.wifiState)
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            println("5G" + wifiManager!!.is5GHzBandSupported)
            println("isDeviceToApRttSupported" + wifiManager!!.isDeviceToApRttSupported)
            println("isEnhancedPowerReportingSupported" + wifiManager!!.isEnhancedPowerReportingSupported)
            println("isP2pSupported" + wifiManager!!.isP2pSupported)
            println("isPreferredNetworkOffloadSupported" + wifiManager!!.isPreferredNetworkOffloadSupported)
            println("isTdlsSupported" + wifiManager!!.isTdlsSupported)
        }
        val scanResults = wifiManager!!.scanResults
        for (result in scanResults) {
            println(result.toString())
        }
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            try {
                val passpointConfigurations = wifiManager!!.passpointConfigurations
                for (passpointConfiguration in passpointConfigurations) {
                    println(passpointConfiguration.toString())
                }
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
    }
}